go-controller/pkg/ovn/loadbalancer/loadbalancer.go

```diff
@@ -252,9 +253,10 @@ func buildLB(lb *LB) *nbdb.LoadBalancer {
 	}
 
 	options := map[string]string{
-		"reject":    reject,
-		"event":     event,
-		"skip_snat": skipSNAT,
+		"reject":          reject,
+		"event":           event,
+		"skip_snat":       skipSNAT,
+		"hairpin_snat_ip": fmt.Sprintf("%s %s", types.V4OVNServiceHairpinMasqueradeIP, types.V6OVNServiceHairpinMasqueradeIP),
```

Comment:
yes we break haha let me fix this as discussed.

Reply:
this also tells me we don't have ANY tests for node -> other node port service case? let me add that to the new e2e

Reply:
err something funny is happening on the GR side of things.. seems like the OVN hairpin isn't working as expected?
```
21:41:31.250391 breth0 Out ifindex 6 02:42:ac:12:00:04 ethertype IPv4 (0x0800), length 80: (tos 0x0, ttl 64, id 47289, offset 0, flags [DF], proto TCP (6), length 60)
    172.18.0.4.33463 > 172.18.0.3.32766: Flags [S], cksum 0x585a (incorrect -> 0x3659), seq 3329178340, win 64240, options [mss 1460,sackOK,TS val 152220014 ecr 0,nop,wscale
 7], length 0
21:41:31.250399 eth0  Out ifindex 35 02:42:ac:12:00:04 ethertype IPv4 (0x0800), length 80: (tos 0x0, ttl 64, id 47289, offset 0, flags [DF], proto TCP (6), length 60)
    172.18.0.4.33463 > 172.18.0.3.32766: Flags [S], cksum 0x585a (incorrect -> 0x3659), seq 3329178340, win 64240, options [mss 1460,sackOK,TS val 152220014 ecr 0,nop,wscale
 7], length 0
21:41:31.250997 eth0  In  ifindex 35 02:42:ac:12:00:03 ethertype IPv4 (0x0800), length 80: (tos 0x0, ttl 63, id 47289, offset 0, flags [DF], proto TCP (6), length 60)
    172.18.0.3.33463 > 172.18.0.4.80: Flags [S], cksum 0xb607 (correct), seq 3329178340, win 64240, options [mss 1460,sackOK,TS val 152220014 ecr 0,nop,wscale 7], length 0
21:41:31.251192 breth0 In  ifindex 6 02:42:ac:12:00:03 ethertype IPv4 (0x0800), length 80: (tos 0x0, ttl 63, id 47289, offset 0, flags [DF], proto TCP (6), length 60)
    172.18.0.3.33463 > 172.18.0.4.80: Flags [S], cksum 0xb607 (correct), seq 3329178340, win 64240, options [mss 1460,sackOK,TS val 152220014 ecr 0,nop,wscale 7], length 0
21:41:31.251212 breth0 Out ifindex 6 02:42:ac:12:00:04 ethertype IPv4 (0x0800), length 80: (tos 0x0, ttl 64, id 0, offset 0, flags [DF], proto TCP (6), length 60)
    172.18.0.4.80 > 172.18.0.3.33463: Flags [S.], cksum 0x585a (incorrect -> 0xabad), seq 2260452723, ack 3329178341, win 65160, options [mss 1460,sackOK,TS val 152220015 ec
r 152220014,nop,wscale 7], length 0
21:41:31.251215 eth0  Out ifindex 35 02:42:ac:12:00:04 ethertype IPv4 (0x0800), length 80: (tos 0x0, ttl 64, id 0, offset 0, flags [DF], proto TCP (6), length 60)
    172.18.0.4.80 > 172.18.0.3.33463: Flags [S.], cksum 0x585a (incorrect -> 0xabad), seq 2260452723, ack 3329178341, win 65160, options [mss 1460,sackOK,TS val 152220015 ec
r 152220014,nop,wscale 7], length 0
21:41:31.251554 eth0  In  ifindex 35 02:42:ac:12:00:03 ethertype IPv4 (0x0800), length 80: (tos 0x0, ttl 63, id 0, offset 0, flags [DF], proto TCP (6), length 60)
    172.18.0.3.32766 > 172.18.0.4.33463: Flags [S.], cksum 0x2bff (correct), seq 2260452723, ack 3329178341, win 65160, options [mss 1460,sackOK,TS val 152220015 ecr 1522200
14,nop,wscale 7], length 0
21:41:31.251558 breth0 In  ifindex 6 02:42:ac:12:00:03 ethertype IPv4 (0x0800), length 80: (tos 0x0, ttl 63, id 0, offset 0, flags [DF], proto TCP (6), length 60)
    172.18.0.3.32766 > 172.18.0.4.33463: Flags [S.], cksum 0x2bff (correct), seq 2260452723, ack 3329178341, win 65160, options [mss 1460,sackOK,TS val 152220015 ecr 1522200
14,nop,wscale 7], length 0
21:41:31.251574 breth0 Out ifindex 6 02:42:ac:12:00:04 ethertype IPv4 (0x0800), length 72: (tos 0x0, ttl 64, id 47290, offset 0, flags [DF], proto TCP (6), length 52)
```
seems like the hairpin doesn't kick in :/

```
_uuid               : 5c4923f0-0270-4745-b51d-de8fd127b6c6
external_ids        : {"k8s.ovn.org/kind"=Service, "k8s.ovn.org/owner"="service-hairpin-test-7887/service-for-pods"}
health_check        : []
ip_port_mappings    : {}
name                : "Service_service-hairpin-test-7887/service-for-pods_TCP_node_router+switch_ovn-worker2"
options             : {event="false", hairpin_snat_ip="169.254.169.5 fd69::5", reject="true", skip_snat="false"}
protocol            : tcp
selection_fields    : []
vips                : {"10.96.210.53:6666"="172.18.0.4:80", "172.18.0.3:32766"="172.18.0.4:80"}
```

Reply:
@trozet : I spoke to @dceara ... it seems `hairpin_snat_ip` will never have an effect on the GR LBs.. because we use the lb_force_snat_ip, so we are protected by default.. blessing in disguise! :D I added an e2e to ensure it works as expected. 

```
_uuid               : 89da5101-0dfe-4fa9-b582-24ba3a647f89
copp                : 8ebdf146-a973-4b18-94ef-d03b34ec487c
enabled             : []
external_ids        : {physical_ip="172.18.0.3", physical_ips="172.18.0.3"}
load_balancer       : [2933e7f7-e839-430a-9fc8-b8f915a356b4, 5c4923f0-0270-4745-b51d-de8fd127b6c6]
load_balancer_group : [dfed3f05-24e5-4473-9d25-43648e0a2265]
name                : GR_ovn-worker2
nat                 : [46d812c8-5074-4897-90d5-d22845ef9279, 5cbca707-4c55-4626-a0a5-be2fcd069aeb, fddc028b-627b-4e62-9081-0ae706b5a88a]
options             : {always_learn_from_arp_request="false", chassis="b884f3e8-d463-45a2-9ada-47fd05758320", dynamic_neigh_routers="true", lb_force_snat_ip=router_ip, snat-ct-zone="0"}
policies            : []
ports               : [8afa7322-1403-4f3c-ae56-27cc7d33910f, bc94a544-9c7d-4ae6-809d-01abbe4dce9b]
static_routes       : [1dc42066-ae7a-443b-889c-337d720bc884, bd143f44-0608-447f-bdf5-35c50b89b768]
```

Reply:
I added an e2e test just to make sure we catch this case well..